### PR TITLE
perl-gd: update to 2.77, update urls

### PR DESCRIPTION
**Pull request by A-N-Other**
Updating to `@2.77`. The url for this package changed a while ago - updated it here.

### CODE REVIEW

File: var/spack/repos/builtin/packages/perl-gd/package.py [hunk 1]
```diff
@@ -10,11 +10,19 @@ class PerlGd(PerlPackage):
     """Interface to Gd Graphics Library"""
 
     homepage = "https://metacpan.org/pod/GD"
-    url = "http://search.cpan.org/CPAN/authors/id/L/LD/LDS/GD-2.53.tar.gz"
+    url = "https://cpan.metacpan.org/authors/id/R/RU/RURBAN/GD-2.77.tar.gz"
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/perl-gd/package.py:13]: You don't need this property when `url_for_version()` is added.  But I can see keeping it if the older version(s) will be deprecated and there's a plan to remove `url_for_version()`.

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/perl-gd/package.py:13]: Let me know if you want to remove it at this time.

**Review by tldahlgren — APPROVED** [review]
Confirmed all version sha256.  Just a quick question about url vs url_for_version.

File: var/spack/repos/builtin/packages/perl-gd/package.py (hunk 1 above)
Comment (A-N-Other) [reply var/spack/repos/builtin/packages/perl-gd/package.py:13]: There are a lot of versions between the two checksummed here, so might be easier to leave it with a mind to deprecating the old one, but shall leave the decision for you ... what's 'canonical'?

Comment (tldahlgren) [reply var/spack/repos/builtin/packages/perl-gd/package.py:13]: >  what's 'canonical'?

Depends on context but basically the standard, archetype, or (if form) natural unique representation.

**Pull request merged**